### PR TITLE
Avijit/infer multi

### PR DESCRIPTION
**Pull request by avijit-nervana**
This is an example of multiple networks that are running in parallel while inferring.

### CODE REVIEW

File: examples/cpp/infer_multiple_networks/inference_engine.cc [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    ngraph::Event::write_trace(read_event);
+    ngraph::Event::write_trace(infer_event);
+    if (m_step_callback != nullptr) {
+      m_step_callback(step_count);
+    }
+    step_count++;
+
+    // Check if we are asked to terminate
+    if (m_terminate_worker) {
+      cout << "[" << m_name << "] m_terminate_worker: SIgnaled" << std::endl;
```
Comment (sayantan-nervana) [inline examples/cpp/infer_multiple_networks/inference_engine.cc]:
```suggestion
      cout << "[" << m_name << "] m_terminate_worker: Signaled" << std::endl;
```

File: examples/cpp/infer_multiple_networks/inference_engine.h [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  Status CreateSession(const string& network, unique_ptr<Session>& session);
+
+  const string m_name;
+  unique_ptr<Session> m_session;
+  void ThreadMain();
+  thread m_worker;
+  atomic<bool> m_terminate_worker{false};
+  std::function<void(int)> m_step_callback{nullptr};
+
+  // Image relatd info
```
Comment (sayantan-nervana) [inline examples/cpp/infer_multiple_networks/inference_engine.h]:
```suggestion
  // Image related info
```

File: examples/cpp/infer_multiple_networks/inference_engine.cc [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+                                      unique_ptr<Session>& session) {
+  SessionOptions options;
+  options.config.mutable_graph_options()
+      ->mutable_optimizer_options()
+      ->set_opt_level(OptimizerOptions_Level_L0);
+  options.config.mutable_graph_options()
+      ->mutable_rewrite_options()
+      ->set_constant_folding(RewriterConfig::OFF);
+
+  // The following is related to Grapller - which we are turning off
```
Comment (sayantan-nervana) [inline examples/cpp/infer_multiple_networks/inference_engine.cc]:
```suggestion
  // The following is related to Grappler - which we are turning off
```

File: examples/cpp/infer_multiple_networks/main.cc [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+  }
+
+  // We need to call this to set up global state for TensorFlow.
+  tensorflow::port::InitMain(argv[0], &argc, &argv);
+  if (argc > 1) {
+    std::cout << "Error: Unknown argument " << argv[1] << "\n" << usage;
+    return -1;
+  }
+
+  // const char* backend = "CPU";
```
Comment (shresthamalik) [inline examples/cpp/infer_multiple_networks/main.cc]: Minor : dead code

**Pull request merged**